opnode/l2/output_derivation.go

```diff
@@ -0,0 +1,35 @@
+package l2
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type BlockPreparer interface {
+	GetPayload(ctx context.Context, payloadId PayloadID) (*ExecutionPayload, error)
+	ForkchoiceUpdated(ctx context.Context, state *ForkchoiceState, attr *PayloadAttributes) (ForkchoiceUpdatedResult, error)
+}
```

Comment:
What's the rationale for separating `BlockPreparer` from `EngineAPI`?

Reply:
Reducing the scope of things to mock in testing (as outlined in the PR description though, I'm not sure how best to test these consecutive interface calls. Maybe we just use testify mock, and catch the calls?

Reply:
But you'll need to mock `ExecutePayload` anyway to test `DriverStep`, so not sure what the advantage is.
(Also, if you merge the two files, the tests can live in a single test file.)

Not sure of the nature of your testing question: can't we just have a mock implementation of `DriverAPI` (with failing implems for `EthBackend` since we don't need them here).